### PR TITLE
Adds `loop` expression

### PR DESCRIPTION
**Pull request by baszalmstra**
Enabling a much faster fibonacci:

```mun
fn fibonacci(n:int):int {
    let a = 0;
    let b = 1;
    let i = 1;
    loop {
        if i > n {
            return a
        }
        let sum = a + b;
        a = b;
        b = sum;
        i += 1;
    }
}
```

**Depends on #49**

### CODE REVIEW

**Comment by baszalmstra** [comment]
[Loop statements](https://trello.com/c/0HNpcSgI/63-loop-statements)

File: crates/mun_runtime/src/test.rs [hunk 1]
```diff
@@ -185,6 +185,57 @@ fn fibonacci() {
     assert_invoke_eq!(i64, 5, driver, "fibonacci", 5i64);
     assert_invoke_eq!(i64, 89, driver, "fibonacci", 11i64);
     assert_invoke_eq!(i64, 987, driver, "fibonacci", 16i64);
+
+    driver.update(
+        r#"
+    fn fibonacci(n:int):int {
```
Comment (Wodann) [inline crates/mun_runtime/src/test.rs]: A duplicate of `fibonacci_loop`?

File: crates/mun_hir/src/ty/infer.rs [hunk 2]
```diff
@@ -278,14 +299,18 @@ impl<'a, D: HirDatabase> InferenceResultBuilder<'a, D> {
...
                 } else if self.return_ty != Ty::Empty {
                     self.diagnostics
                         .push(InferenceDiagnostic::ReturnMissingExpression { id: tgt_expr });
                 }
 
                 Ty::simple(TypeCtor::Never)
             }
+            Expr::Loop { body } => {
+                self.infer_expr(*body, &Expectation::has_type(Ty::Empty));
+                Ty::simple(TypeCtor::Never)
```
[the remark this answers is not in the document]
Comment (baszalmstra) [reply crates/mun_hir/src/ty/infer.rs]: The return type is the the type of the expression. Both `loop` and `return` never return a type, that's why they have the `Never` return type.

[the remark this answers is not in the document]
Comment (Wodann) [reply crates/mun_hir/src/ty/infer.rs]: Okay. So do `return` or `break` statements in the loop actually turn into variable assignments?

**Pull request merged**